File: Kernel Simulator/Network/FTP/Commands/Put.vb

'    Kernel Simulator  Copyright (C) 2018-2022  EoflaOE
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports KS.Network.FTP.Transfer

Namespace Network.FTP.Commands
    Class FTP_PutCommand
        Inherits CommandExecutor
        Implements ICommand

        Public Overrides Sub Execute(StringArgs As String, ListArgs() As String, ListArgsOnly As String(), ListSwitchesOnly As String()) Implements ICommand.Execute
            Dim LocalFile As String = ListArgs(0)
            Dim RemoteFile As String = If(ListArgs.Count > 1, ListArgs(1), "")
            TextWriterColor.Write(DoTranslation("Uploading file {0}..."), False, ColTypes.Progress, ListArgs(0))
            Dim Result As Boolean = If(Not String.IsNullOrWhiteSpace(LocalFile), FTPUploadFile(RemoteFile, LocalFile), FTPUploadFile(RemoteFile))
            If Result Then
                Console.WriteLine()
                TextWriterColor.Write(DoTranslation("Uploaded file {0}"), True, ColTypes.Success, LocalFile)
            Else
                Console.WriteLine()
                TextWriterColor.Write(DoTranslation("Failed to upload {0}"), True, ColTypes.Error, LocalFile)
            End If
        End Sub

    End Class
End Namespace

File: Kernel Simulator/Network/FTP/Commands/Put.cs
using System.Linq;
using KS.ConsoleBase.Colors;
using KS.Languages;
using KS.Misc.Writers.ConsoleWriters;

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using KS.Network.FTP.Transfer;
using KS.Shell.ShellBase.Commands;

namespace KS.Network.FTP.Commands
{
	class FTP_PutCommand : CommandExecutor, ICommand
	{

		public override void Execute(string StringArgs, string[] ListArgs, string[] ListArgsOnly, string[] ListSwitchesOnly)
		{
			string LocalFile = ListArgs[0];
			string RemoteFile = ListArgs.Count() > 1 ? ListArgs[1] : "";
			TextWriterColor.Write(Translate.DoTranslation("Uploading file {0}..."), false, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Progress), ListArgs[0]);
			bool Result = !string.IsNullOrWhiteSpace(LocalFile) ? FTPTransfer.FTPUploadFile(RemoteFile, LocalFile) : FTPTransfer.FTPUploadFile(RemoteFile);
			if (Result)
			{
				TextWriterColor.WritePlain("", true);
				TextWriterColor.Write(Translate.DoTranslation("Uploaded file {0}"), true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Success), LocalFile);
			}
			else
			{
				TextWriterColor.WritePlain("", true);
				TextWriterColor.Write(Translate.DoTranslation("Failed to upload {0}"), true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Error), LocalFile);
			}
		}

	}
}